### PR TITLE
[POR-2199] Refactor DB get, list and delete endpoints

### PR DESCRIPTION
**Pull request by ferozemohideen**
These endpoints were previously reliant on passing cloud provider information into requests; however, since the front-end does not currently have a generalized concept of cloud provider or organization, and since databases are tied to clusters currently, it makes sense to store cloud provider details in an external data source. This PR refactors the 3 endpoints listed above to use the datastores table as a source of truth (which is especially helpful in the `list` endpoint, since the helm list is super slow).

I will need to seed the prod db with currently existing databases before these changes are pushed to prod, otherwise current dashboards will break.

UPDATE: the prod db has been seeded as of 1/17 at 10 PM ET

### CODE REVIEW

**Review by stefanmcshane — APPROVED** [review]

File: dashboard/src/main/home/database-dashboard/forms/DatabaseForm.tsx [hunk 1]
```diff
@@ -60,39 +66,80 @@ const DatabaseForm: React.FC<Props> = ({
...
+                placeholder="ex: academic-sophon-db"
+                type="text"
+                width="300px"
+                error={errors.name?.message}
+                {...register("name")}
+              />
+              {currentProject?.multi_cluster && (
+                <>
+                  <Spacer y={1} />
+                  <Selector<string>
```
Comment (ianedwards) [inline dashboard/src/main/home/database-dashboard/forms/DatabaseForm.tsx:115]: maybe add a description here explaining why they need to choose a cluster

File: internal/datastore/delete.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+// DeleteDatastoreRecordInput is the input type for DeleteDatastoreRecord
+type DeleteDatastoreRecordInput struct {
+	ProjectID uint
+	Name      string
+
+	DatastoreRepository repository.DatastoreRepository
+}
+
+// DeleteDatastoreRecord deletes a datastore record by name
+func DeleteDatastoreRecord(ctx context.Context, inp DeleteDatastoreRecordInput) (*models.Datastore, error) {
```
Comment (ianedwards) [inline internal/datastore/delete.go:21]: for these methods in the datastores package, you could just name them by the operation so other packages call something that looks like

```
datastores.Get()
datastores.List()
datastores.Delete()
```

File: api/server/handlers/cloud_provider/list_aws.go [hunk 3]
```diff
@@ -34,6 +35,14 @@ type AwsAccount struct {
 	ProjectID uint `json:"project_id"`
 }
 
+// CloudProvider is an abstraction for a cloud provider
+type CloudProvider struct {
+	// Type is the type of the cloud provider
+	Type porterv1.EnumCloudProvider `json:"type"`
```
[the remark this answers is not in the document]
Comment (ferozemohideen) [reply api/server/handlers/cloud_provider/list_aws.go:41]: good point. Right now all the ccp db calls take in this enum, so I would have to add a couple switch statements to go back and forth. We're gonna refactor this and probably move this all to ccp anyway

File: dashboard/src/lib/hooks/useDatabaseList.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      );
+
+      const parsed = await datastoreListResponseValidator.parseAsync(
+        response.data
+      );
+      return parsed.datastores;
+    },
+    {
+      enabled: !!currentProject?.id && currentProject.id !== -1,
+      refetchInterval: 10000,
```
[the remark this answers is not in the document]
Comment (ferozemohideen) [reply dashboard/src/lib/hooks/useDatabaseList.ts:41]: good call

**Pull request merged**